### PR TITLE
ISPN-5412 Native Spark Connector

### PR DESCRIPTION
**Pull request by ghost**
https://issues.jboss.org/browse/ISPN-5412

### CODE REVIEW

**Comment by ghost** [comment]
PR Updated.

File: src/test/scala/org/infinispan/spark/ClusteredSuites.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.infinispan.spark.suites.{DistributedSuite, ReplicatedSuite, SQLSuite, StreamingSuite}
+import org.infinispan.spark.test.Cluster
+import org.scalatest.{BeforeAndAfterAll, Suites}
+
+import scala.language.postfixOps
+
+/**
+ * Aggregates all suites that requires a running cluster.
+ */
+class ClusteredSuites extends Suites(new DistributedSuite, new ReplicatedSuite, new SQLSuite, new StreamingSuite) with BeforeAndAfterAll {
```
Comment (galderz) [inline src/test/scala/org/infinispan/spark/ClusteredSuites.scala:12]: Great use of `BeforeAndAfterAll` to improve efficiency of testsuite, :clap:

**Comment by galderz** [comment]
It looks good to me. Had a brief discussion with Gustavo about how the client listeners are implemented here. They bring both keys and values for all data which when might not perform as well as if the filtering and conversion were directly done in the server.

Anyone else wants to have a look into this?

**Comment by wburns** [comment]
@galderz When I looked over it, it seemed fine to me as well.  Although I don't know a ton about the inner workings here let alone scala :)

I agree regarding listeners, that would definitely be preferred.

**Comment by ghost** [comment]
yes, the `InfinispanInputDStream` should be more flexible to allow filtering at the source rather than only when inside Spark, I'll create a JIRA for that. In any case, it is not a blocker.

**Comment by galderz** [comment]
@gustavonalle Actually, to make the module more usable, I'd suggest adding an `sbt` script to the root of the project so that people are not forced to install `sbt` in advance. See [here](https://github.com/paulp/sbt-extras/blob/master/sbt).

**Comment by ghost** [comment]
@galderz ok, will add it :+1:

**Comment by ghost** [comment]
@galderz Added the sbt launcher

**Comment by galderz** [comment]
Thanks @gustavonalle, pushed to master :)

**Pull request closed without merging**